src/main/java/com/atlassian/db/replica/api/PessimisticPropagationConsistency.java

```diff
@@ -14,13 +14,13 @@
  * Assumes that writes propagate from main to replicas in at most a given amount of time.
  * If it cannot remember the time of last write, pessimistically assumes it's going to be inconsistent.
  */
-public class PessimisticPropagationConsistency implements ReplicaConsistency {
+public final class PessimisticPropagationConsistency implements ReplicaConsistency {
```

Comment:
Technically this breaks source compatibility. Subclasses will fail compilation now. So we should mark it in the changelog and bump a major version.

Reply:
IMO we should leave it as is. There's no need to change it now.

Reply:
Kotlin made me lazy, forgot that I have to opt-in to maintainability keywords in Java 😉